www/docs/server/infer-types.md

```diff
@@ -16,46 +46,79 @@ It is often useful to wrap functionality of your `@trpc/client` or `@trpc/react`
 
 Let's assume we have this example router:
 
-```ts title='routers/_app.ts'
-export const appRouter = t.router({
-  user: t.router({
-    me: t.procedure.query(() => {
-      // ...
-      return { id: 1, name: 'John' };
-    }),
-  }),
-  post: t.procedure.input(z.string()).mutation(() => {
-    // ...
-    return post.id;
-    }),
-});
-
-export type AppRouter = typeof appRouter;
+```ts twoslash title='server.ts'
+// @include: server
 ```
 
 By traversing the router object, you can infer the types of the procedures. The following example shows how to infer the types of the procedures of the example `appRouter`:
 
-```ts title='types.ts'
-import type { 
-  inferProcedureOutput, 
-  inferProcedureInput, 
-} from '@trpc/server';
-import type { AppRouter } from 'routers/_app';
+```ts twoslash title="client.ts"
+// @module: esnext
+// @include: server
+// ---cut---
+// @filename: client.ts
+import type { inferProcedureInput, inferProcedureOutput } from '@trpc/server';
+import type { AppRouter } from './server';
+
+// @noErrors
+type Post = inferProcedureOutput<AppRouter['post']['byId']>;
+//   ^?
+type PostInput = inferProcedureInput<AppRouter['post']['create']>;
+//   ^?
+```
+
+### Additional DX Helper Type
+
+If you don't like the double-import from the above snippet, `@trpc/server` also exports a type `GetInferenceHelpers<TRouter, 'input' | 'output'>`. This lets you pass your router once at initialization, then import a single helper type when inferring types:
+
+```ts twoslash title='utils/trpc.ts'
+// @include: server
+// ---cut---
+import type { GetInferenceHelpers } from '@trpc/server';
+import type { AppRouter } from './server';
 
-export type User = inferProcedureOutput<AppRouter['user']['me']>;
-//           ^? { id: number, name: string }}
+// @noErrors
+export type InferProcedureInput = GetInferenceHelpers<AppRouter, 'input'>;
+export type InferProcedureOutput = GetInferenceHelpers<AppRouter, 'output'>;
+```
 
-export type PostInput = inferProcedureInput<AppRouter['post']>;
-//           ^? string
+<!-- FIXME: reuse above snippet -->
+```ts twoslash
+// @module: esnext
+// @include: server
+// @filename: utils.ts
+import type { GetInferenceHelpers } from '@trpc/server';
+import type { AppRouter } from './server';
+
+export type InferProcedureInput = GetInferenceHelpers<AppRouter, 'input'>;
+export type InferProcedureOutput = GetInferenceHelpers<AppRouter, 'output'>;
+// @filename: index.ts
+// ---cut---
+import { InferProcedureInput, InferProcedureOutput } from './utils';
+
+type Post = InferProcedureOutput['post']['byId'];
+//   ^?
+type PostInput = InferProcedureInput['post']['create'];
+//   ^?
 ```
 
+
+
 ## Infer `TRPClientError`s based on your router
 
-```ts title='src/client.ts'
-import type {
-  AppRouter
-} from '~/server/routers/_app';
+```ts twoslash title='client.ts'
```

Comment:
fixed the twoslash for this but not sure if we want it since it's a very large and complex type that doesn't really provide much insight?